### PR TITLE
Raise API server errors as Java exceptions

### PR DESCRIPTION
**Pull request by cdonati**
Without this, the application error field on remote api responses are ignored.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/5914/

File: AppServer_Java/src/com/google/appengine/tools/development/ApiServer.java [hunk 1]
```diff
@@ -131,6 +133,8 @@ public void close() {
         boolean parsed = response.mergeFrom(post.getResponseBodyAsStream());
         if (!parsed) {
             throw new IOException("Error parsing the response from the HTTP API server.");
+        } else if (response.hasApplicationError()) {
+            throw ApiProxyUtils.getRpcError(packageName, methodName, StatusProto.getDefaultInstance(), response.getApplicationError().getCode(), response.getApplicationError().getDetail());
```
Comment (sjones4) [inline AppServer_Java/src/com/google/appengine/tools/development/ApiServer.java]: I think the following gives equivalent functionality but avoids the additional dependency:

`throw new ApplicationException(response.getApplicationError().getCode(), response.getApplicationError().getDetail());`

Comment (cdonati) [reply AppServer_Java/src/com/google/appengine/tools/development/ApiServer.java]: Wow, that's a huge cleanup. Thanks!

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/6008/
https://ocd.appscale.com:8080/job/Autoscale-test-manual-triggers/2317/
https://ocd.appscale.com:8080/job/Taskqueue-e2e-Test/392/

**Pull request merged**